internal/nodes/avi_model_evh_nodes.go

```diff
@@ -503,75 +503,35 @@ func (v *AviEvhVsNode) GetNodeType() string {
 
 func (o *AviEvhVsNode) AddFQDNAliasesToHTTPPolicy(host string, hosts []string, key string) {
 
-	var hppMap *AviHostPathPortPoolPG
-	var redirectPorts *AviRedirectPort
-
 	// Find the hppMap and redirectPorts that matches the host
 	for _, policy := range o.HttpPolicyRefs {
 		for j := range policy.HppMap {
-			if utils.HasElem(policy.HppMap[j].Host, host) {
-				hppMap = &policy.HppMap[j]
-				break
-			}
+			policy.HppMap[j].Host = make([]string, len(hosts))
+			copy(policy.HppMap[j].Host, hosts)
 		}
 		for j := range policy.RedirectPorts {
-			if utils.HasElem(policy.RedirectPorts[j].Hosts, host) {
-				redirectPorts = &policy.RedirectPorts[j]
-				break
-			}
-		}
-		if utils.HasElem(policy.AviMarkers.Host, host) {
-			for _, host := range hosts {
-				if !utils.HasElem(policy.AviMarkers.Host, host) {
-					policy.AviMarkers.Host = append(policy.AviMarkers.Host, host)
-				}
-			}
-		}
-	}
-
-	// Update the hppMap with the hosts
-	if hppMap != nil {
-		for _, host := range hosts {
-			if !utils.HasElem(hppMap.Host, host) {
-				hppMap.Host = append(hppMap.Host, host)
-			}
-		}
-	}
-
-	// Update the redirectPorts with the hosts
-	if redirectPorts != nil {
-		for _, host := range hosts {
-			if !utils.HasElem(redirectPorts.Hosts, host) {
-				redirectPorts.Hosts = append(redirectPorts.Hosts, host)
-			}
+			policy.RedirectPorts[j].Hosts = make([]string, len(hosts))
+			copy(policy.RedirectPorts[j].Hosts, hosts)
 		}
+		policy.AviMarkers.Host = make([]string, len(hosts))
+		copy(policy.AviMarkers.Host, hosts)
 	}
 
 	utils.AviLog.Debugf("key: %s, msg: Added hosts %v to HTTP policy for VS %s", key, hosts, o.Name)
 }
 
 func (o *AviEvhVsNode) RemoveFQDNAliasesFromHTTPPolicy(host string, hosts []string, key string) {
 
-	// Find the hppMap and redirectPorts that matches the host and remove the hosts
-	for _, policy := range o.HttpPolicyRefs {
-		for j := range policy.HppMap {
-			if utils.HasElem(policy.HppMap[j].Host, host) {
-				for _, host := range hosts {
-					policy.HppMap[j].Host = utils.Remove(policy.HppMap[j].Host, host)
-				}
-			}
-		}
-		for j := range policy.RedirectPorts {
-			if utils.HasElem(policy.RedirectPorts[j].Hosts, host) {
-				for _, host := range hosts {
-					policy.RedirectPorts[j].Hosts = utils.Remove(policy.RedirectPorts[j].Hosts, host)
-				}
+	for _, host := range hosts {
```

Comment:
the `host` variable is conflicting with the function argument.

Reply:
I have removed the function parameter `host` as it is not used in the function.